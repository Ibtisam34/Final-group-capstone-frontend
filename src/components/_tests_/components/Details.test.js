import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Detail from '../../Details';

const doctor = {
  image: 'https://img.freepik.com/free-vector/doctor-character-background_1270-84.jpg?w=2000',
  name: 'Dr. John Doe',
  specialization: 'Cardiology',
  email: 'john.doe@example.com',
  phone: '+1234567890',
  availability: 'Mon-Fri 9AM - 5PM',
  id: '1',
};

const isLoggedIn = true;

describe('Detail page', () => {
  test('it renders doctor name in details page', () => {
    render(
      <Provider store={store}>
        <Router>
          <Detail doctor={doctor} isLoggedIn={isLoggedIn} />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Dr. John Doe')).toBeInTheDocument();
  });

  it('renders correctly', () => {
    window.localStorage.setItem('userId', '1');
    const tree = render(
      <Provider store={store}>
        <Router>
          <Detail doctor={doctor} isLoggedIn={isLoggedIn} />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
