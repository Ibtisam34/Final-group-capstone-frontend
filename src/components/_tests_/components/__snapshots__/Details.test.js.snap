// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="details-container"
        >
          <div
            class="flex-direction"
          >
            <div>
              <img
                alt="doctor"
                class="thumbnail images rounded"
                src="https://img.freepik.com/free-vector/doctor-character-background_1270-84.jpg?w=2000"
              />
            </div>
            <div
              class="card-body"
            >
              <div
                class="details-text-container"
              >
                <div>
                  <h3
                    class="text-dark"
                  >
                    Dr. John Doe
                  </h3>
                  <p
                    class="description my-4"
                  >
                    Cardiology
                  </p>
                  <div>
                    <p
                      class="description"
                    >
                      Mon-Fri 9AM - 5PM
                    </p>
                    <p
                      class="description"
                    >
                      Email:
                       
                      john.doe@example.com
                    </p>
                    <p
                      class="description"
                    >
                      Phone:
                       
                      +1234567890
                    </p>
                  </div>
                  <div>
                    <a
                      href="/undefined/1/appointment"
                    >
                      <button
                        class="btn-success theme-btn mt-4 rounded my-btn"
                        type="button"
                      >
                        Book
                      </button>
                    </a>
                  </div>
                  <div>
                    <a
                      href="/undefined"
                    >
                      <svg
                        class="mt-4"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="details-container"
      >
        <div
          class="flex-direction"
        >
          <div>
            <img
              alt="doctor"
              class="thumbnail images rounded"
              src="https://img.freepik.com/free-vector/doctor-character-background_1270-84.jpg?w=2000"
            />
          </div>
          <div
            class="card-body"
          >
            <div
              class="details-text-container"
            >
              <div>
                <h3
                  class="text-dark"
                >
                  Dr. John Doe
                </h3>
                <p
                  class="description my-4"
                >
                  Cardiology
                </p>
                <div>
                  <p
                    class="description"
                  >
                    Mon-Fri 9AM - 5PM
                  </p>
                  <p
                    class="description"
                  >
                    Email:
                     
                    john.doe@example.com
                  </p>
                  <p
                    class="description"
                  >
                    Phone:
                     
                    +1234567890
                  </p>
                </div>
                <div>
                  <a
                    href="/undefined/1/appointment"
                  >
                    <button
                      class="btn-success theme-btn mt-4 rounded my-btn"
                      type="button"
                    >
                      Book
                    </button>
                  </a>
                </div>
                <div>
                  <a
                    href="/undefined"
                  >
                    <svg
                      class="mt-4"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
